packages/svelte-check/src/writers.ts

```diff
@@ -0,0 +1,124 @@
+            this._stream.write(`${chalk.green('File')} : ${chalk.green(filename)}\n`);
+
+            diagnostics.forEach((diagnostic) => {
+                const source = diagnostic.source ? `(${diagnostic.source})` : '';
+
+                // eslint-disable-next-line max-len
+                const position = `Line: ${diagnostic.range.start.line}, Character: ${diagnostic.range.start.character}`;
+
+                // Show some context around diagnostic range
+                const startOffset = offsetAt(diagnostic.range.start, text);
+                const endOffset = offsetAt(diagnostic.range.end, text);
+                const codePrev = chalk.cyan(text.substring(Math.max(startOffset - 10, 0), startOffset));
+                const codeHighlight = chalk.magenta(text.substring(startOffset, endOffset));
+                const codePost = chalk.cyan(text.substring(endOffset, endOffset + 10));
+                const code = codePrev + codeHighlight + codePost;
+                let msg;
+
+                if (this._isVerbose) {
+                    msg = `${diagnostic.message} ${source}\n${position}\n${chalk.cyan(code)}`;
+                }
+                else {
+                    msg = diagnostic.message;
+                }
+
+                if (diagnostic.severity === DiagnosticSeverity.Error) {
+                    this._stream.write(`${chalk.red('Error')}: ${msg}\n`);
+                }
+                else {
+                    this._stream.write(`${chalk.yellow('Warn')}: ${msg}\n`);
+                }
+            });
+
+            this._stream.write('\n');
+        }
+    }
+
+    completion(_fileCount: number, errorCount: number, _warningCount: number) {
+        this._stream.write('====================================\n');
+
+        if (errorCount === 0) {
+            this._stream.write(chalk.green(`svelte-check found no errors\n`));
+        } else {
+            this._stream.write(chalk.red(`svelte-check found ${errorCount} ${errorCount === 1 ? 'error' : 'errors'}\n`));
+        }
+    }
+
+    failure(err: Error) {
+        this._stream.write(`${err}\n`);
+    }
+}
+
+export class MachineFriendlyWriter implements Writer {
+    _stream: Writable;
+
+    constructor(stream: Writable) {
+        this._stream = stream;
+    }
+
+    _log(msg: string) {
+        this._stream.write(`${new Date().getTime()} ${msg}\n`);
+    }
+
+    start(workspaceUri: string) {
+        this._log(`START ${JSON.stringify(workspaceUri)}`);
+    }
+
+    file(diagnostics: Diagnostic[], filename: string, _text: string) {
+        diagnostics.forEach((d) => {
+            const { message, severity, range } = d;
+            const type = severity === DiagnosticSeverity.Error ? "ERROR" : severity === DiagnosticSeverity.Warning ? "WARNING" : null;
+
+            if (type) {
+                const { line, character } = range.start;
+                this._log(`${type} ${JSON.stringify(filename)} ${line}:${character} ${JSON.stringify(message)}`);
```

Comment:
Just curious: Why do you use `JSON.stringify` here? If it's to have `"` around the content, maybe just do `"${filename}"` instead.

Reply:
That's to escape special characters, e.g. quotes